ibis/expr/operations.py

```diff
@@ -696,6 +708,14 @@ class Mean(Reduction):
     output_type = rules.scalar_output(_mean_output_type)
 
 
+class Quantile(Reduction):
+
+    input_type = [value,
+                  number(name='quantile', allow_boolean=False),
+                  string(name='interpolation', optional=True)]
```

Comment:
I think we want to use `rules.string_options(list_of_possible_interpolation_values)` so that we can limit these before executing.